DiscordBot/Commands/EventCommands.cs

```diff
@@ -217,6 +221,89 @@ public async Task EditEvent(CommandContext context, InteractivityModule interact
             await EditEventField(context, interactivity, eventKey.Value, editField, eventEmbed);
         }
 
+        private static async Task CreateSignupSheet(CommandContext context, InteractivityModule interactivity)
+        {
+            await context.RespondAsync($"{context.Member.Mention} - what is the event key? (use the ``list`` option to find out)");
+            var eventKey = await GetUserIntResponse(context, interactivity);
+            if (eventKey == null)
+            {
+                return;
+            }
+
+            var eventEmbed = await GetEventEmbed(context, eventKey.Value);
+            if (eventEmbed == null)
+            {
+                return;
+            }
+
+            await context.RespondAsync($"{context.Member.Mention} - start signups for this event? - (``yes``/``no``)", embed: eventEmbed);
+            var confirmationResponse = await GetUserConfirmation(context, interactivity);
+            if (confirmationResponse == null || confirmationResponse == false)
+            {
+                return;
+            }
+
+            await SendSignupMessage(context, eventKey.Value);
+        }
+
+        private static async Task SendSignupMessage(CommandContext context, int eventKey)
+        { 
+            var eventsSheetsService = context.Dependencies.GetDependency<IEventsSheetsService>();
+            var discordEvent = await eventsSheetsService.GetEventAsync(eventKey);
+
+            var userResponseDictionary = await eventsSheetsService.GetSignupsByUserAsync(eventKey);
+            var optionsList = await eventsSheetsService.GetSignupsByResponseAsync(eventKey);
```

Comment:
Actually do you need to get both of these? They both contain all the responses and all the users right? I don't think it's worth doing two requests to get the same information? 🙂

Reply:
Only using userResponseDictionary won't because response types with no signups won't be shown.

Only using optionsList would could work but the structure isn't ideal for displaying users alongside their respective responses, meaning I'd have to create something like the userResponseDictionary from optionsList.

Reply:
I don't think we should do both requests though, that seems wasteful. Because it would mean doing two google sheets requests I believe, correct me if I'm wrong @benjimarshall. If the returned structure isn't ideal then maybe the service api needs tweaking?